### PR TITLE
Mutli Resource Visits Entries- 2415

### LINKED ISSUE #2415: Report Detail: Most visited resource count is increased by 2
Most Visited Resources count increased by 2 rather than 1 on single visit to the particular resource.

### PR DESCRIPTION
**Pull request by nadaa**
fixes #2415 

I solved this by adding a variable to control updating the resource.

### CODE REVIEW

**Comment by lmmrssa** [comment]
@nadaa It is still updating resource_activities for image. For other it is updating once but for image its still updating twice.

**Comment by nadaa** [comment]
@lmmrssa, yes, this is not the optimal solution.

File: src/app/resources/view-resources/resources-viewer.component.ts [hunk 1]
```diff
@@ -37,12 +38,15 @@ export class ResourcesViewerComponent implements OnChanges, OnDestroy {
   ) {
     this.resourcesService.resourcesListener(this.parent).pipe(takeUntil(this.onDestroy$))
       .subscribe((resources) => {
+        this.isResource = false;
```
Comment (paulbert) [inline src/app/resources/view-resources/resources-viewer.component.ts]: Save resource id here (maybe as `shownResourceId`)

File: src/app/resources/view-resources/resources-viewer.component.ts [hunk 2]
```diff
@@ -37,12 +38,15 @@ export class ResourcesViewerComponent implements OnChanges, OnDestroy {
...
         this.setResource(resources.find((r: any) => r._id === this.resourceId));
       });
   }
 
   ngOnChanges() {
-    this.resourcesService.requestResourcesUpdate(this.parent, this.fetchRating);
+    if (this.isResource) {
```
Comment (paulbert) [inline src/app/resources/view-resources/resources-viewer.component.ts]: Check if resource id equals `this.resourceId` here

**Comment by paulbert** [comment]
Ooops, I gave bad advice before.  Since `requestResourceUpdate` is started asynchronously in the parent component, `shownResourceId` is not set before the initial `ngOnChanges` and the activity is still logged twice (or three times in some of my testing).

Needed to check after the request completes.

**Comment by nadaa** [comment]
@paulbert, I checked before submitting, it was working!

**Pull request merged**